benchmark/tests/test_main.py

```
@@ -255,6 +258,19 @@ def close(self):
 
 
+def test_read_optional_token_fails_fast_for_missing_file(tmp_path):
+    with pytest.raises(ValueError, match="unable to read benchmark token file"):
+        main._read_optional_token(tmp_path / "missing-token")
+
+
+def test_read_optional_token_fails_fast_for_empty_file(tmp_path):
+    token_file = tmp_path / "control_token"
+    token_file.write_text("  \n", encoding="utf-8")
+
+    with pytest.raises(ValueError, match="benchmark token file .* is empty"):
```

Ruff (0.15.18)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)